pythonx/jedi_vim.py

```diff
@@ -134,6 +134,58 @@ def echo_highlight(msg):
     sys.path.remove(parso_path)
 
 
+class VimCompat:
+    _eval_cache = {}
+    _func_cache = {}
+
+    @classmethod
+    def has(cls, what):
+        try:
+            return cls._eval_cache[what]
+        except KeyError:
+            ret = cls._eval_cache[what] = cls.call('has', what)
+            return ret
+
+    @classmethod
+    def call(cls, func, *args):
+        try:
+            f = cls._func_cache[func]
+        except KeyError:
+            if IS_NVIM:
+                f = cls._func_cache[func] = getattr(vim.funcs, func)
+            else:
+                f = cls._func_cache[func] = vim.Function(func)
+        return f(*args)
```

Comment:
This should / needs to be checked to how it behaves to the current method of going through `jedi#_vim_exceptions` / `_catch_exception`.
Would be necessary to get the "throwpoint" it seems.

Reply:
Do you expect me to comment on this? Because I don't really understand it :) Which is fine of course, just asking.

Reply:
I've wondered if it is better/worse than the previous method, but then tested it.  I think it is good for now.